halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java

```diff
@@ -107,21 +106,21 @@ public void serialize(Object value, JsonGenerator gen, SerializerProvider provid
         };
     }
 
-    protected StdScalarSerializer<Object> getSecretFileSerializer(BeanPropertyWriter beanPropertyWriter, SecretFile annotation) {
+    protected StdScalarSerializer<Object> getSecretFileSerializer(BeanPropertyWriter beanPropertyWriter, SecretFile annotation, boolean shouldDecrypt) {
         return new StdScalarSerializer<Object>(String.class, false) {
             @Override
             public void serialize(Object value, JsonGenerator gen, SerializerProvider provider) throws IOException {
                 if (value != null) {
                     String sValue = value.toString();
-                    if (EncryptedSecret.isEncryptedSecret(sValue)) {
+                    if (!EncryptedSecret.isEncryptedSecret(sValue)) {
+                        sValue = annotation.prefix() + sValue;
+                    } else if (shouldDecrypt) {
                         // Decrypt the content of the file and store on the profile under a random
                         // generated file name
                         String name = newRandomFilePath(beanPropertyWriter.getName());
                         byte[] bytes = secretSessionManager.decryptAsBytes(sValue);
                         profile.getDecryptedFiles().put(name, bytes);
                         sValue = getCompleteFilePath(name);
```

Comment:
Can this simply be `sValue = annotation.prefix() + getCompleteFilePath(name);` ?

Reply:
yes, will change!